bin/puppet-validator

```diff
@@ -89,5 +102,24 @@ else
   logger.level     = loglevel
   options[:logger] = logger
 
-  PuppetValidator.run! options
+  if ssl_opts[:cert_chain_file] and ssl_opts[:private_key_file]
+    options[:ssl] = true
+  end
+
+  # These options should either both be nil or both be Strings
+  unless ssl_opts[:cert_chain_file].class == ssl_opts[:private_key_file].class
+    raise 'You must specify both the certificate and key file!'
+  end
+
+  PuppetValidator.run!(options) do |server|
+    if options[:ssl]
+      if server.respond_to? 'ssl='
```

Comment:
I'm yet not familiar with how Sinatra works, but I assume that this `respond_to` construct is something that it supports and that it takes care of creating the self-signed cert automatically, if needed?

Reply:
The `.respond_to` is a rubyism that introspects an object to see if it has the named method. It's not Sinatra-specific. This particular method (`ssl=`) is checking to see if the server has the methods that allow me to set ssl options in this way.

If it *does*, then the server is known to accept the parameters as defined and will either use the certs specified or auto-generate a self-signed cert.

Reply:
Excellent, thank you